extensions/graphviz_support/graphviz_support/generate_graphviz/generate_graphviz.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# Copyright 2004-present Facebook. All Rights Reserve
+
+import labgraph as lg
+from labgraph.graphs.stream import Stream
+from graphviz import Digraph
+from typing import List
+from ..method.method import Method
+from ..errors.errors import GenerateGraphvizError
+
+
+def identify_graph_nodes(graph: lg.Graph) -> List[Method]:
+    """
+    Function that identifies the graph methods
+    and return them as individual nodes
+
+    @params:
+        graph: instance of the running computational graph
+
+    @return: List of nodes(methods)
+    """
+    nodes: List[Method] = []
+
```

Comment:
The data structure for graph construction in this PR can be simplified.
For example, we can use indegrees, and outdegrees.
The nodes with zero indegrees will be the root for the graph.
Function find_connections can be further simplified as well. We can use DFS for the graph constructions.
Since this would be a Directed Acyclic Graph (DAG), we just need to store the downstream nodes (__out_adjacents).

Reply:
I don't see how a DFS can be directly applied here. I'm using Topics & Streams to build the downstream nodes list. Please! correct me in case I'm missing something

Reply:
An pseudocode reference can be found below (but the eventual implementation will depends on the graph structure). Currently, find_connections use three nested loops, can we reduce the time complexity by reducing the number of nested loops?
```
def example():
        graph = collections.defauldict([])
        indegrees = collections.defauldict(0)
        outdegrees = collections.defauldict(0)
        
        # build graph and count indegress/outdegrees
        for u, v in input:
            # directed edge u -> v
            graph[u].append(v)
            # v as incoming connection
            indegrees[v] += 1 
            # u has outgoing connection 
            outdegrees[u] += 1
        
        # unconnected verticies have outdegree 0
        for i in range(1, N + 1):
            if i not in outdegrees:
                outdegrees[i] = 0 
```